File: app/src/main/java/io/intrepid/contest/models/Category.java
package io.intrepid.contest.models;

import android.os.Parcel;
import android.os.Parcelable;

public class Category implements Parcelable {
    private String name;
    private String description;

    public Category(String name, String description) {
        this.name = name;
        this.description = description;
    }

    protected Category(Parcel in) {
        name = in.readString();
        description = in.readString();
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(name);
        dest.writeString(description);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<Category> CREATOR = new Creator<Category>() {
        @Override
        public Category createFromParcel(Parcel in) {
            return new Category(in);
        }

        @Override
        public Category[] newArray(int size) {
            return new Category[size];
        }
    };

    @Override
    public boolean equals(Object obj) {
        try {
            Category otherCategory = (Category) obj;
            return this.name.equals(otherCategory.name) &&
                    this.description.equals(otherCategory.description);
        } catch (ClassCastException exception) {
            return false;
        }
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }
}

File: app/src/main/java/io/intrepid/contest/models/Category.kt
package io.intrepid.contest.models

import android.os.Parcel
import android.os.Parcelable
import java.util.*

class Category : Parcelable {
    val id: UUID? = null
    var name: String? = null
    var description: String? = null

    constructor(name: String, description: String) {
        this.name = name
        this.description = description
    }

    protected constructor(`in`: Parcel) {
        name = `in`.readString()
        description = `in`.readString()
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(name)
        dest.writeString(description)
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun equals(obj: Any?): Boolean {
        try {
            val otherCategory = obj as Category?
            return this.name == otherCategory!!.name && this.description == otherCategory.description
        } catch (exception: ClassCastException) {
            return false
        }
    }

    companion object {
        val CREATOR: Parcelable.Creator<Category> = object : Parcelable.Creator<Category> {
            override fun createFromParcel(`in`: Parcel): Category {
                return Category(`in`)
            }

            override fun newArray(size: Int): Array<Category?> {
                return arrayOfNulls(size)
            }
        }
    }
}
